src/Microsoft.Identity.Client/Platforms/Shared/Apple/CoreCustomUrlProtocol.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Foundation;
+
+namespace Microsoft.Identity.Client.Platforms.Shared.Apple
+{
+    internal class CoreCustomUrlProtocol : NSUrlProtocol
+    {
+        private NSUrlConnection connection;
+
+        [Export("canInitWithRequest:")]
+        public static bool canInitWithRequest(NSUrlRequest request)
+        {
+            if (request.Url.Scheme.Equals("https", StringComparison.CurrentCultureIgnoreCase))
+            {
+                return GetProperty("ADURLProtocol", request) == null;
+            }
+
+            return false;
+        }
+
+        [Export("canonicalRequestForRequest:")]
+        public new static NSUrlRequest GetCanonicalRequest(NSUrlRequest request)
+        {
+            return request;
+        }
+
+        [Export("initWithRequest:cachedResponse:client:")]
+        public CoreCustomUrlProtocol(
+            NSUrlRequest request,
+            NSCachedUrlResponse cachedResponse,
+            INSUrlProtocolClient client)
+            : base(request, cachedResponse, client)
+        {
+        }
+
+        public override void StartLoading()
+        {
+            if (this.Request == null)
+            {
+                return;
+            }
+
+            NSMutableUrlRequest mutableRequest = (NSMutableUrlRequest) this.Request.MutableCopy();
+            SetProperty(new NSString("YES"), "ADURLProtocol", mutableRequest);
+            this.connection = new NSUrlConnection(mutableRequest, new CoreCustomConnectionDelegate(this), true);
+        }
+
+        public override void StopLoading()
+        {
+            this.connection.Cancel();
```

Comment:
null check?

Reply:
This is a refactoring, I don't plan to make changes here, but I'll add the null check.